### PR TITLE
 automated: linux: wireguard fioconfig test

### PR DESCRIPTION
**Pull request by brenwoo**
Signed-off-by: Brendan Wood <brendan.wood@foundries.io>

### CODE REVIEW

File: automated/linux/wireguard-fioconfig/wg-pk.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+! check_root && error_msg "You need to be root to run this script."
+create_out_dir "${OUTPUT}"
+cd "${OUTPUT}" || exit
+
+#check /run/secrets exists, create if not
+[ -d /run/secrets ] || mkdir /run/secrets
+
+echo "endpoint=217.44.46.222:5186" > /run/secrets/wireguard-server
+echo "server_address=10.42.42.1" >> /run/secrets/wireguard-server
```
[the remark this answers is not in the document]
Comment (brenwoo) [reply automated/linux/wireguard-fioconfig/wg-pk.sh:18]: The contents of the file only exist to be overwritten, added a comment highlighting this.

File: automated/linux/wireguard-fioconfig/wg-pk.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    echo "Waiting for fioconfig to start"
+    sleep 1
+done
+
+m2=$(md5sum "$wgServerFile")
+
+if [ "$m1" !=  "$m2" ] ; then
+    echo "Wireguard-server updated." >& 2
+    report_pass "wg-pk-test"
+    exit 1
```
Comment (nareshkamboju) [inline automated/linux/wireguard-fioconfig/wg-pk.sh:38]: exit 1 
not needed here.

File: automated/linux/wireguard-fioconfig/wg-pk.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+else
+    report_fail "wg-pk-test"
```
Comment (nareshkamboju) [inline automated/linux/wireguard-fioconfig/wg-pk.sh:40]: you may have
exit 1
here since it is fail case.

**Review by nareshkamboju — CHANGES_REQUESTED** [review]

File: automated/linux/wireguard-fioconfig/wg-pk.sh (hunk 3 above)
Comment (brenwoo) [reply automated/linux/wireguard-fioconfig/wg-pk.sh:40]: Changed. Thanks!

**Pull request merged**